implementation/src/main/java/io/smallrye/mutiny/converters/uni/UniToMultiPublisher.java

```diff
@@ -78,7 +87,8 @@ public void onItem(T item) {
 
         @Override
         public void onFailure(Throwable failure) {
-            if (upstream.getAndSet(CANCELLED) != CANCELLED) {
+            if (STATE_UPDATER.compareAndSet(this, UniToMultiSubscription.State.UNI_REQUESTED,
```

Comment:
We can get a failure before the request (the consumer gets the subscription and a failure before requesting).
The same can apply to completion.

Reply:
My bad, you only subscribe to the uni if you get a request, so it's good.

Reply:
Thanks I'll merge in an hour or so.